src/EditorFeatures/Core/Extensibility/Navigation/NavigableItemFactory.DeclaredSymbolNavigableItem.cs

```diff
@@ -53,6 +51,22 @@ public DeclaredSymbolNavigableItem(Document document, DeclaredSymbolInfo declare
                     }
                 });
             }
+
+            private ISymbol FindSymbol()
+            {
+                // Here, we will use partial semantics. We are going to use this symbol to get a glyph, display string,
+                // and potentially documentation comments. The first two should work fine even if we don't have full
+                // references, and the latter will probably be fine. (It wouldn't be if you have a partial type
+                // and we didn't get all the trees parsed yet to know that. In other words, an edge case we don't care about.)
```

Comment:
If the symbol is a member (field/property/method/etc), then the types it references may not exist in the partial solution, so the display string may end up incorrectly displaying things as error types instead of their real types.  It may still be worth the optimization, but the comment should probably call it out as a known issue, and it's something we should watch for when using this change to see if it's a problem in practice.